src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -728,208 +751,266 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
+        /**
+         * @private
+         * @param e {Event} The event received from the `<video>` element
+         */
+        onVolumeChange = (e) => {
+            /**
+             * @type {HTMLMediaElement}
+             */
+            const elem = e.target;
+            htmlMediaHelper.saveVolume(elem.volume);
+            events.trigger(this, 'volumechange');
         }
 
-        function onNavigatedToOsd() {
-            var dlg = videoDialog;
+        /**
+         * @private
+         */
+        onNavigatedToOsd = () => {
+            const dlg = this.videoDialog;
             if (dlg) {
                 dlg.classList.remove('videoPlayerContainer-onTop');
 
-                onStartedAndNavigatedToOsd();
+                this.onStartedAndNavigatedToOsd();
             }
         }
 
-        function onStartedAndNavigatedToOsd() {
+        /**
+         * @private
+         */
+        onStartedAndNavigatedToOsd() {
             // If this causes a failure during navigation we end up in an awkward UI state
-            setCurrentTrackElement(subtitleTrackIndexToSetOnPlaying);
+            this.setCurrentTrackElement(this.subtitleTrackIndexToSetOnPlaying);
 
-            if (audioTrackIndexToSetOnPlaying != null && self.canSetAudioStreamIndex()) {
-                self.setAudioStreamIndex(audioTrackIndexToSetOnPlaying);
+            if (this.audioTrackIndexToSetOnPlaying != null && this.canSetAudioStreamIndex()) {
+                this.setAudioStreamIndex(this.audioTrackIndexToSetOnPlaying);
             }
         }
 
-        function onPlaying(e) {
-            if (!self._started) {
-                self._started = true;
-                this.removeAttribute('controls');
+        /**
+         * @private
+         * @param e {Event} The event received from the `<video>` element
+         */
+        onPlaying = (e) => {
+            /**
+             * @type {HTMLMediaElement}
+             */
+            const elem = e.target;
+            if (!this._started) {
+                this._started = true;
+                elem.removeAttribute('controls');
 
                 loading.hide();
 
-                htmlMediaHelper.seekOnPlaybackStart(self, e.target, self._currentPlayOptions.playerStartPositionTicks, function () {
-                    if (currentSubtitlesOctopus) {
-                        currentSubtitlesOctopus.timeOffset = (self._currentPlayOptions.transcodingOffsetTicks || 0) / 10000000 + currentTrackOffset;
-                        currentSubtitlesOctopus.resize();
-                        currentSubtitlesOctopus.resetRenderAheadCache(false);
+                htmlMediaHelper.seekOnPlaybackStart(this, e.target, this._currentPlayOptions.playerStartPositionTicks, function () {
+                    if (this.currentSubtitlesOctopus) {
+                        this.currentSubtitlesOctopus.timeOffset = (this._currentPlayOptions.transcodingOffsetTicks || 0) / 10000000 + this.currentTrackOffset;
```

Comment:
```suggestion
                        this.currentSubtitlesOctopus.timeOffset = (this._currentPlayOptions.transcodingOffsetTicks || 0) / 10000000 + this.currentTrackOffset();
```

Reply:
this needs to be changed

Reply:
This isn't a method:
```javascript
resetSubtitleOffset() {
    this.currentTrackOffset = 0;
    this.showTrackOffset = false;
}
```

Reply:
Oh yeah, my mistake. I've been staring at github for too long :P